src/generic/internet-connection-alert/index.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { Warning as WarningIcon } from '@edx/paragon/icons';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import AlertMessage from '../alert-message';
+import messages from './messages';
+
+const InternetConnectionAlert = ({
+  isQueryPending, dispatchMethod, onInternetConnectionFailed, onDispatchMethodCall,
+}) => {
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const [showAlert, setShowAlert] = useState(false);
+  const [isOnline, setIsOnline] = useState(window.navigator.onLine);
+
+  useEffect(() => {
+    const handleOnlineStatus = () => setIsOnline(window.navigator.onLine);
+
+    window.addEventListener('online', handleOnlineStatus);
+    window.addEventListener('offline', handleOnlineStatus);
+
+    return () => {
+      window.removeEventListener('online', handleOnlineStatus);
+      window.removeEventListener('offline', handleOnlineStatus);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (isQueryPending && !isOnline) {
+      setShowAlert(true);
+      onInternetConnectionFailed();
+    } else if (isQueryPending && isOnline) {
+      dispatch(dispatchMethod);
+      onDispatchMethodCall();
+      setShowAlert(false);
+    }
+  }, [isOnline, isQueryPending]);
+
+  useEffect(() => {
```

Comment:
why didn't you use the hook above as this one makes the same logic and depends on the same property `isOnline`?

Reply:
it looks like this code was used for debugging and it is not needed, removed thanks!